source/utilities.tex

```diff
@@ -19492,3 +19493,1887 @@
+when binary or hexadecimal output is used,
+this option adds the respective prefix
+\tcode{0b} (\tcode{0B}) or
+\tcode{0x} (\tcode{0X})
+to the output value.
+For nonzero integers,
+when octal output is used,
+this option adds the prefix \tcode{0}
+to the output value.
+Whether the prefix is lower-case or upper-case
+is determined by the case of the type format specifier.
+For floating-point numbers
+the alternate form causes the result of the conversion
+to always contain a decimal-point character,
+even if no digits follow it.
+% FIXME: This is a weird place for this part of the spec to appear.
+Normally, a decimal-point character appears in the result of these
+conversions only if a digit follows it.
+In addition, for \tcode{g} and \tcode{G} conversions,
+% FIXME: Are they normally?
+trailing zeros are not removed from the result.
+
+\pnum
+% FIXME: What if it's an arg-id?
+The \fmtgrammarterm{positive-integer} in
+\fmtgrammarterm{width} is a decimal integer defining the minimum field width.
+If \fmtgrammarterm{width} is not specified,
+there is no minimum field width, and
+the field width is determined based on the content of the field.
+
+\pnum
+A zero (\tcode{0}) character
+preceding the \fmtgrammarterm{width} field
+pads the field with leading zeros (following any indication of sign or base)
+to the field width,
+except when applied to an infinity or NaN.
+This option is only valid for
+arithmetic types other than \tcode{charT} and \tcode{bool}
+or when an integer presentation type is specified.
+If the \tcode{0} character and an \fmtgrammarterm{align} option both appear,
+the \tcode{0} character is ignored.
+\begin{example}
+\begin{codeblock}
+char c = 120;
+string s1 = format("{:+06d}", c);       // value of \tcode{s1} is \tcode{"+00120"}
+string s2 = format("{:#06x}", 0xa);     // value of \tcode{s2} is \tcode{"0x000a"}
+string s3 = format("{:<06}", -42);      // value of \tcode{s3} is \tcode{"-42\ \ \ "} (\tcode{0} is ignored because of \tcode{<} alignment)
+\end{codeblock}
+\end{example}
+
+\pnum
+% FIXME: What if it's an arg-id?
+The \fmtgrammarterm{nonnegative-integer} in
+\fmtgrammarterm{precision} is a decimal integer defining
+the precision or maximum field size.
+It can only be used with floating-point and string types.
+For floating-point types this field specifies the formatting precision.
+For string types it specifies how many characters will be used from the string.
+
+\pnum
+The \fmtgrammarterm{type} determines how the data should be presented.
+
+\pnum
+% FIXME: What is a "string" here, exactly?
+The available string presentation types are specified in \tref{format.type.string}.
+%
+\begin{floattable}{Meaning of \fmtgrammarterm{type} options for strings}{format.type.string}{ll}
+\topline
+\lhdr{Type} & \rhdr{Meaning} \\ \rowsep
+none, \tcode{s} &
+Copies the string to the output.
+\\
+\end{floattable}
+
+\pnum
+% FIXME: What about void* and nullptr_t?
+% FIXME: What about strings? Do they get padding for alignment?
```

Comment:
Nothing suggests they don't, and `fill-and-align` isn't limited to some types unlike `sign` and `0`, which specifies that they do.

Reply:
This wording explicitly says when and how the alignment padding is added, and it's not unreasonable to interpret this as saying this is the *only* time that such padding is added (since nowhere else actually says to do so). Eg, someone could ask "Why would the standard list this rule here if it could say nothing and get the same result?" (This is one reason why it's dangerous to duplicate normative wording: it creates the -- often unintended -- implication that the scope of the prior wording is being restricted to the duplicate case.)

I'm going to convert the end of the paragraph to a note so that it doesn't create this incorrect impression.

Reply:
Fixed.